File: sbe/binance/spot_3_1/generated-php/spot_sbe/ErrorResponse.php
<?php
/**
 * Generated SBE (Simple Binary Encoding) message codec.
 */

class ErrorResponse
{
    public const TEMPLATE_ID = 100;
    public const SCHEMA_ID = 3;
    public const SCHEMA_VERSION = 1;
    public const BLOCK_LENGTH = 18;

    public int|float|array|null $code = null;
    public int|float|array|null $serverTime = null;
    public int|float|array|null $retryAfter = null;
    public string $msg = '';
    public string $data = '';

    private function decodeVarData(string $data, int &$offset): string
    {
        $length = unpack('V', substr($data, $offset, 4))[1];
        $offset += 4;
        $value = substr($data, $offset, $length);
        $offset += $length;
        return $value;
    }

    public function encode(): string
    {
        $buffer = '';

        if ($this->code !== null) {
            $buffer .= pack('s', $this->code);
        }
        if ($this->serverTime !== null) {
            $buffer .= pack('q', $this->serverTime);
        }
        if ($this->retryAfter !== null) {
            $buffer .= pack('q', $this->retryAfter);
        }

        return $buffer;
    }

    public function decode(string $data): void
    {
        $offset = 0;

        $this->code = unpack('s', substr($data, $offset, 2))[1];
        $offset += 2;
        $this->serverTime = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;
        $this->retryAfter = unpack('q', substr($data, $offset, 8))[1];
        $offset += 8;

        // Skip to end of block for forward compatibility
        $offset = 18;


        $this->msg = $this->decodeVarData($data, $offset);
        $this->data = $this->decodeVarData($data, $offset);
    }
}

File: sbe/binance/spot_3_1/generated-python/spot_sbe/ErrorResponse.py
"""Generated SBE (Simple Binary Encoding) message codec."""

import struct
from typing import List, Optional, Tuple, Set
from io import BytesIO

class ErrorResponse:
    """SBE message: ErrorResponse."""

    TEMPLATE_ID = 100
    SCHEMA_ID = 3
    SCHEMA_VERSION = 1
    BLOCK_LENGTH = 18

    def __init__(self):
        self.code = None
        self.server_time = None
        self.retry_after = None
        self.msg = b''
        self.data = b''

    def _decode_var_data(self, data: bytes, offset: int) -> Tuple[bytes, int]:
        """Decode variable-length binary data."""
        pos = offset
        length = struct.unpack_from('<I', data, pos)[0]
        pos += 4
        value = data[pos:pos+length]
        pos += length
        return (value, pos)

    def encode(self) -> bytes:
        """Encode the message to bytes."""
        buffer = BytesIO()

        if self.code is not None:
            buffer.write(struct.pack('<h', self.code))
        if self.server_time is not None:
            buffer.write(struct.pack('<q', self.server_time))
        if self.retry_after is not None:
            buffer.write(struct.pack('<q', self.retry_after))

        return buffer.getvalue()

    def decode(self, data: bytes) -> None:
        """Decode the message from bytes."""
        pos = 0

        self.code = struct.unpack_from('<h', data, pos)[0]
        pos += 2
        self.server_time = struct.unpack_from('<q', data, pos)[0]
        pos += 8
        self.retry_after = struct.unpack_from('<q', data, pos)[0]
        pos += 8

        # Skip to end of block for forward compatibility
        pos = 18


        self.msg, pos = self._decode_var_data(data, pos)
        self.data, pos = self._decode_var_data(data, pos)
